lib/fluent/root_agent.rb

```diff
@@ -47,24 +48,57 @@ module Fluent
   class RootAgent < Agent
     ERROR_LABEL = "@ERROR".freeze # @ERROR is built-in error label
 
-    def initialize(log:, system_config: SystemConfig.new)
+    class SourceOnlyMode
+      DISABELD = 0
+      NORMAL = 1
+      RESTART_WITHOUT_DOWNTIME_READY_ONLY = 2
+
+      def initialize(with_source_only, start_in_parallel)
+        if start_in_parallel
+          @mode = RESTART_WITHOUT_DOWNTIME_READY_ONLY
+        elsif with_source_only
+          @mode = NORMAL
+        else
+          @mode = DISABELD
+        end
+      end
+
+      def source_only?
```

Comment:
I prefer `enabled?` or `available?` as this method name

Reply:
Thanks! It would be more concise.

Reply:
Fixed.